src/sick_mrs1000_parser.cpp

```diff
@@ -0,0 +1,346 @@
+/*
```

Comment:
One general comment on this file: There's one major change we would like to propose. It would be better to publish the point cloud for each layer separately immediately after parsing the scan instead of accumulating 4 scans into one point cloud. This is under the assumption that the scanner doesn't record the 4 layers in parallel, but sequentially.

This would have the following advantages:

* **Most importantly:** The time stamps would be more correct, since each layer gets its own time stamp.
* Also, it would simplify the code a lot: you wouldn't need `point_counter_` and `layer_count_` any more and wouldn't have to do the ugly check in line 241.
* The `cloud_` variable wouldn't need to be a member variable any more.
* We can rely on the `config_` variable not suddenly changing, since we do all the processing in one function call.

Reply:
After discussion with @spuetz last week, I've changed my mind. Our assumption that the layers were recorded sequentially was false, they all are recorded in parallel and only transferred sequentially. So it makes sense to publish the accumulated point cloud. Just ignore this comment.